### ISSUE TITLE
Issues with 3D Bounding Box Detection Accuracy in Custom Dope Model

### ISSUE THREAD

**Issue by phsilvarepo** [NVlabs/Deep_Object_Pose#376]
Hello, 

I am currently trying to develop a custom Dope model to detect a custom .obj model. I am struggling to get good results on the inference side. I have generated a 75k dataset, ant trained with train.py from train2 folder on the current repository for 100 epochs. Although the loss as decreased the inference seems to be struggling to accurately detect the dimension of the 3D bounding box. I am not sure what could be improved to attain better results, perhaps increase the scale of the object in the generated dataset. 

In the inference side I have added the files in the common folder to the train2 due to not recognizing the cuboid module, and also commented `make_belief_debug_img=True` in line 264 in inference.py. The center of the object is detected relatively well but the remaining bounding box does not perform well.

.obj model:
[m6.zip](https://github.com/user-attachments/files/16392920/m6.zip)

Ground truth:
![000018](https://github.com/user-attachments/assets/2f67016c-8142-4fc1-b6fa-66b12c9ffab0)
![000028](https://github.com/user-attachments/assets/884975d3-e0ae-4f90-97bf-2a801bf6b209)
![000034](https://github.com/user-attachments/assets/5a944770-dae3-419d-950d-d97b11b9e384)
![000035](https://github.com/user-attachments/assets/1195fa12-b0d1-4c4f-9749-97bc82ff4852)
![000038](https://github.com/user-attachments/assets/0a36205e-1393-4142-9650-166caafb955a)

Training run:
![Screenshot from 2024-07-26 14-54-15](https://github.com/user-attachments/assets/454ee11c-4ae2-4f90-8ffb-5120de2f7a38)

Config file:
```
topic_camera: "/dope/webcam/image_raw"
topic_camera_info: "/dope/webcam/camera_info"
topic_publishing: "dope"
input_is_rectified: True   # Whether the input image is rectified (strongly suggested!)
downscale_height: 400      # if the input image is larger than this, scale it down to this pixel height

# Comment any of these lines to prevent detection / pose estimation of that object
weights: {
    'drill_6':"/home/rics/catkin_ws/src/Deep_Object_Pose/train2/tmp/net_epoch_100.pth"
}

# Type of neural network architecture
architectures: {
    'drill_6':"dope",
}


# Cuboid dimension in cm x,y,z
dimensions: {
    "drill_6": [1.14,1.14,11.3],
}

class_ids: {
    "drill_6": 1,
}

draw_colors: {
    "drill_6": [13, 255, 128],  # green
}

# optional: provide a transform that is applied to the pose returned by DOPE
model_transforms: {
}

# optional: if you provide a mesh of the object here, a mesh marker will be
# published for visualization in RViz
# You can use the nvdu_ycb tool to download the meshes: https://github.com/NVIDIA/Dataset_Utilities#nvdu_ycb
meshes: {
}

# optional: If the specified meshes are not in meters, provide a scale here (e.g. if the mesh is in centimeters, scale should be 0.01). default scale: 1.0.
mesh_scales: {
}

# Config params for DOPE
thresh_angle: 0.5
thresh_map: 0.0001
sigma: 3
thresh_points: 0.1
```

Inference:
![000017](https://github.com/user-attachments/assets/b35803b6-13e3-4104-a305-8340b5e12e2c)
![000024](https://github.com/user-attachments/assets/277d0580-ab34-48cc-8c8b-bb52927a7534)
![000034](https://github.com/user-attachments/assets/6a2392fc-d341-4a1d-93b8-60997b3aa90d)
![000035](https://github.com/user-attachments/assets/0dc329c9-1ad4-485c-b37a-357a4a097386)
![000062](https://github.com/user-attachments/assets/7ff96f30-06ab-47b0-876f-a0f3b21e66ac)

I also attempted to adapt the parameters in camera_info.yaml but the results did not change. 
Any help would be great. Thanks.

**Comment by Manu752** [comment]
Hi, Did you add the cube around your object in the training data set ? this is how my training data set looks like 
![](trash:///Screenshot%20from%202025-06-01%2015-47-45.png)
